### PR TITLE
Radio Receiver

### PR DESCRIPTION
**Pull request by VivaanBahl**

### CODE REVIEW

File: LPC Projects/pwmdemodulator/src/pwmdemodulator.c [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+	}
+
+	//Board_UARTPutSTR(msg_obj.data);
+	Board_UARTPutSTR("\n");
+	int received = 0;
+	received |= (uint8_t)msg_obj.data[0];
+	received |= ((uint8_t)msg_obj.data[1] << 8);
+	received |= ((uint8_t)msg_obj.data[2] << 16);
+	received |= ((uint8_t)msg_obj.data[3] << 24);
+	//char buf[100];
```
Comment (VivaanBahl) [inline LPC Projects/pwmdemodulator/src/pwmdemodulator.c]: delete or refactor

File: LPC Projects/pwmdemodulator/src/pwmdemodulator.c [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+	{
+		RiseTime = CurrentTimer;
+	}
+	else
+	{
+		if (RiseTime != 0)
+		{
+			if (CurrentTimer > k_min_pulse && CurrentTimer < k_max_pulse)
+			{
+				//char buf[100];
```
Comment (VivaanBahl) [inline LPC Projects/pwmdemodulator/src/pwmdemodulator.c]: delete or refactor

File: LPC Projects/pwmdemodulator/src/pwmdemodulator.c [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+    Board_LED_Set(0, true);
+    baudrateCalculate(TEST_CCAN_BAUD_RATE, CanApiClkInitTable);
+
+    Chip_GPIO_Init(LPC_GPIO);
+    // Input pins for the receiver
+    Chip_GPIO_SetPinDIRInput(LPC_GPIO, 0, 2);
+    Chip_GPIO_SetPinDIRInput(LPC_GPIO, 1, 8);
+    Chip_GPIO_SetPinDIRInput(LPC_GPIO, 1, 5);
+
+    // Uart crap
```
Comment (VivaanBahl) [inline LPC Projects/pwmdemodulator/src/pwmdemodulator.c]: watch ur language young man

File: LPC Projects/pwmdemodulator/src/pwmdemodulator.c [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+void CAN_tx(uint8_t msg_obj_num) {
+	Board_UARTPutSTR("Sent\n");
+}
+
+void CAN_rx(uint8_t msg_obj_num) {
+	/* Determine which CAN message has been received */
+	msg_obj.msgobj = msg_obj_num;
+	/* Now load up the msg_obj structure with the CAN message */
+	LPC_CCAN_API->can_receive(&msg_obj);
+	if (msg_obj_num == 1) {
```
Comment (ramensandwich) [inline LPC Projects/pwmdemodulator/src/pwmdemodulator.c]: Do we want to echo every CAN message we receive?

File: LPC Projects/pwmdemodulator/src/pwmdemodulator.c [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+	Chip_IOCON_PinMuxSet(LPC_IOCON, IOCON_PIO1_7, (IOCON_FUNC1 | IOCON_MODE_INACT));/* TXD */
+	Chip_UART_Init(LPC_USART);
+	Chip_UART_SetBaud(LPC_USART, 115200);
+	Chip_UART_ConfigData(LPC_USART, (UART_LCR_WLEN8 | UART_LCR_SBS_1BIT));
+	Chip_UART_SetupFIFOS(LPC_USART, (UART_FCR_FIFO_EN | UART_FCR_TRG_LEV2));
+	Chip_UART_TXEnable(LPC_USART);
+
+	Board_UARTPutSTR("asdfadsfadfasdf\n");
+
+	// CAN crap
```
Comment (ramensandwich) [inline LPC Projects/pwmdemodulator/src/pwmdemodulator.c]: More foul language :/

File: LPC Projects/pwmdemodulator/src/pwmdemodulator.c [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+    // Uart crap
+	Chip_IOCON_PinMuxSet(LPC_IOCON, IOCON_PIO1_6, (IOCON_FUNC1 | IOCON_MODE_INACT));/* RXD */
+	Chip_IOCON_PinMuxSet(LPC_IOCON, IOCON_PIO1_7, (IOCON_FUNC1 | IOCON_MODE_INACT));/* TXD */
+	Chip_UART_Init(LPC_USART);
+	Chip_UART_SetBaud(LPC_USART, 115200);
+	Chip_UART_ConfigData(LPC_USART, (UART_LCR_WLEN8 | UART_LCR_SBS_1BIT));
+	Chip_UART_SetupFIFOS(LPC_USART, (UART_FCR_FIFO_EN | UART_FCR_TRG_LEV2));
+	Chip_UART_TXEnable(LPC_USART);
+
+	Board_UARTPutSTR("asdfadsfadfasdf\n");
```
Comment (ramensandwich) [inline LPC Projects/pwmdemodulator/src/pwmdemodulator.c]: Is this supposed to be a startup message?

File: LPC Projects/pwmdemodulator/src/pwmdemodulator.c (hunk 4 above)
Comment (VivaanBahl) [reply LPC Projects/pwmdemodulator/src/pwmdemodulator.c]: oops that was my bad, forgot to remove this

File: LPC Projects/pwmdemodulator/src/pwmdemodulator.c (hunk 6 above)
Comment (VivaanBahl) [reply LPC Projects/pwmdemodulator/src/pwmdemodulator.c]: yes I will change that

File: LPC Projects/pwmdemodulator/src/pwmdemodulator.c [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+CCAN_MSG_OBJ_T msg_obj;
+
+uint16_t k_min_pulse = 500;
+uint16_t k_max_pulse = 2500;
+uint16_t k_offset_rc_in = 1500;
+uint16_t k_scale_rc_in = 350;
+uint16_t k_offset_stored_angle = 0;
+uint16_t k_scale_stored_angle = 1500;
+
+uint16_t RiseTime = 0;
```
Comment (ramensandwich) [inline LPC Projects/pwmdemodulator/src/pwmdemodulator.c]: Please rename to SteeringRiseTime to be consistent with the other variables

File: LPC Projects/pwmdemodulator/src/pwmdemodulator.c [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+	uint16_t CurrentTimer = LPC_TIMER32_0->CR[0];
+	if (CurrentValue)
+	{
+		AutonRiseTime = CurrentTimer;
+	}
+	else
+	{
+		if (AutonRiseTime != 0)
+		{
+			auton = CurrentTimer > 1750 ? true : false;
```
Comment (ramensandwich) [inline LPC Projects/pwmdemodulator/src/pwmdemodulator.c]: Why not just "auton = CurrentTime > 1750"?

File: LPC Projects/pwmdemodulator/src/pwmdemodulator.c [hunk 9]
```diff
@@ -0,0 +1,286 @@
...
+	uint16_t CurrentTimer = LPC_TIMER16_1->CR[0];
+	if (CurrentValue)
+	{
+		BrakeRiseTime = CurrentTimer;
+	}
+	else
+	{
+		if (BrakeRiseTime != 0)
+		{
+			brake = CurrentTimer > 1750 ? true : false;
```
Comment (ramensandwich) [inline LPC Projects/pwmdemodulator/src/pwmdemodulator.c]: Why not just `brake = CurrentTime > 1750`?

Comment (ramensandwich) [reply LPC Projects/pwmdemodulator/src/pwmdemodulator.c]: Also, please use a `#define` for 1750 to get rid of the magic number

**Pull request merged**